### PR TITLE
Closes #146 & Closes #147: Integrates find in page

### LINKED ISSUE #146: FNX-5046 ⁃ Find in page UI
Parent #118

### LINKED ISSUE #147: FNX-5052 ⁃ Integrate find in page component
Parent #118

### PR DESCRIPTION
**Pull request by sblatz**

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/toolbar/Toolbar.kt [hunk 1]
```diff
@@ -109,7 +110,9 @@ class Toolbar(
                 context.getString(R.string.browser_menu_find_in_page),
                 R.color.icons
             ) {
-                // TODO Find in Page
+                FindInPageIntegration.launch?.invoke()
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/Toolbar.kt]: We're going to want to use a better solution in the long run than static methods for this. I'll accept this for now, but we may want to use a DI framework like Koin.

**Comment by jonalmeida** [comment]
There's a lifetime bug with FIP right now that can cause the browser to crash. Might be better to wait until to tomorrow to land this?

> lol it compiles

😆

**Comment by sblatz** [comment]
Thanks for the heads up, @jonalmeida!

**Comment by jonalmeida** [comment]
To give an update: there were some parts of find in page that needed reworking in components, so this ended up taking longer than expected. When it lands this can be merged.

**Comment by pocmo** [comment]
> When it lands this can be merged.

Small correction: When that lands it need to be refactored a bit first. Once the PR is merged I'll update the Reference Browser PR to follow that too. :)

Should happen today..

**Comment by jonalmeida** [comment]
@sblatz this is no longer blocked.

The r-b PR which you can use: https://github.com/mozilla-mobile/reference-browser/pull/524

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 2]
```diff
@@ -86,6 +87,16 @@ class BrowserFragment : Fragment() {
             engineView
         )
 
+        val findInPageIntegration = FindInPageIntegration(requireComponents.core.sessionManager, findInPageView)
+        val toolbarIntegration = ToolbarIntegration(
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: This doesn't look right. There's a `ToolbarIntegration` on line 112 in the `lifecycle.addObservers`. If we're adding it here, we should put this reference in the addObservers call below and remove the inlined one.

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 3]
```diff
@@ -86,6 +87,16 @@ class BrowserFragment : Fragment() {
...
+                requireContext(),
+                toolbar,
+                requireComponents.toolbar.shippedDomainsProvider,
+                requireComponents.core.historyStorage
+            )
+
+        lifecycle.addObservers(findInPageIntegration, toolbarIntegration, sessionFeature)
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: We have a `lifecycle.addObservers` for for these classes below. the `findInPageIntegration` should be added to the one below and this deleted.

**Pull request merged**